components/style/selector_impl.rs

```diff
@@ -8,6 +8,73 @@ use selectors::parser::{ParserContext, SelectorImpl};
 pub enum PseudoElement {
     Before,
     After,
+    FirstLine,
+
+    // https://mxr.mozilla.org/mozilla-central/source/layout/style/nsCSSAnonBoxList.h
```

Comment:
Re overkill, I’m also open to changing more significantly how `Stylist` / `PerPseudoElementSelectorMap` / `SelectorMap` are organized. (Though I don’t have a specific design in mind right now.)

Re feature-gated impls, ok, I see what you mean. I’m not necessarily opposed to this, but it doesn’t seem nice. I’d like to explore alternatives some more.